src/audio/rtnr/rtnr.c

```diff
@@ -396,17 +479,149 @@ static int rtnr_cmd_get_data(struct comp_dev *dev,
 	return ret;
 }
 
-static int rtnr_cmd_set_data(struct comp_dev *dev,
+static int rtnr_reconfigure(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	uint8_t *config;
+	size_t size;
+
+	comp_dbg(dev, "rtnr_reconfigure()");
+
+	config = comp_get_data_blob(cd->model_handler, &size, NULL);
+	comp_dbg(dev, "rtnr_reconfigure() size: %d", size);
+
+	if (size == 0) {
+		/* No data to be handled */
+		return 0;
+	}
+
+	if (!config) {
+		comp_err(dev, "rtnr_reconfigure(): Config not set");
+		return -EINVAL;
+	}
+
+	comp_info(dev, "rtnr_reconfigure(): New data applied %p (%zu bytes)",
+		  config, size);
+
+	cd->reconfigure = false;
+
+	RTKMA_API_Set(cd->rtk_agl, config, size, RTNR_DATA_ID_PRESET);
+
+	return 0;
+}
+
+static int rtnr_set_comp_config_by_ipc_config_process(struct comp_dev *dev,
```

Comment:
I would expect that `rtnr_set_comp_config_by_ipc_config_process()` and `rtnr_set_comp_config_by_ipc_ctrl_data()` can be merged as they basically did the same thing. They confused me in the first place with the thought that they were utilized by rtnr_config and rtnr_data respectively.